src/terminal_ui.cc

```diff
@@ -1477,7 +1486,17 @@ void TerminalUI::set_ui_options(const Options& options)
     auto to_bool = [](StringView s) { return s == "yes" or s == "true"; };
 
     m_status_on_top = find("terminal_status_on_top").map(to_bool).value_or(false);
-    m_set_title = find("terminal_set_title").map(to_bool).value_or(true);
+
+    auto title_config = find("terminal_set_title").value_or("yes");
+    if (title_config != "yes" and title_config != "no" and title_config != "true" and title_config != "false")
+    {
+        m_set_title = true;
+        m_title_line = String(title_config);
+    }
+    else {
+        m_set_title = to_bool(title_config);
+	    m_title_line = "";
```

Comment:
nitpick: Indentation is off by one character here

Reply:
Fixed that, thanks!